plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php

```diff
@@ -210,6 +210,29 @@ protected function get_order_item_data( $item ) {
 
 		// Expand meta_data to include user-friendly values.
 		$formatted_meta_data = $item->get_all_formatted_meta_data( null );
+
+		// Filter out product variations.
+		if ( isset( $product ) && 'true' === $this->request['order_item_display_meta'] ) {
+			$order_item_name   = $data['name'];
+			$data['meta_data'] = array_filter(
+				$data['meta_data'],
+				function( $meta ) use ( $product, $order_item_name ) {
+					$meta->key     = rawurldecode( (string) $meta->key );
+					$meta->value   = rawurldecode( (string) $meta->value );
```

Comment:
We're mutating these properties but only when *order_item_display_meta* is *'true'.* I don't know that this will actually cause problems, but it introduces an inconsistency.

Might it be better to either mutate copies of these properties (ie, `$key = rawurldecode(...)`) or else **always** overwrite them with url decoded values?

Reply:
Yes, good call!